Migrated from `setup.py` to `pyproject.toml`

AVDiv opened the pull request:
Replaced the `setup.py` with `pyproject.toml`.

Now the project could be installed using the `python3 -m build` command.

Related Issue: #119

vprusso commented:
Hi @AVDiv ,

Thanks for the contribution! 

A few minor comments, but it's looking good:

1. It looks like `setup.py.old` should be removed.
2. Is it possible to install this package via `poetry install` -> `poetry shell`? I think this would just require a `[tool.poetry]` heading with a few elements in the `pyproject.toml` file. 

Thanks again!

AVDiv commented:
I'll work on it

AVDiv commented:
- [x] Removed old setup file (`setup.py`)
- [x] Changed the build system to poetry, now the package is installable using the `poetry install` command.

vprusso commented:
Wonderful. I was able to `poetry install` -> `poetry shell` and it works great. 

As one final check, let me just ping @WrathfulSpatula to see if there's anything else I might be missing, but for me LGTM! Great work, @AVDiv , and I hope to see more contributions from you.

WrathfulSpatula reviewed (approved):
LGTM!

AVDiv commented:
Welcome @vprusso, but this PR is not being listed on the unitaryhack leader-board, Can you please check it out

vprusso commented:
You are correct. I assigned you to the issue itself. Thanks for letting me know!
